aim/storage/query.py

```diff
@@ -91,6 +96,29 @@ def __call__(
                           metric_name=metric_name)
 
 
+class RunMetadataCache:
+    def __init__(
+            self,
+            repo: 'Repo'
+    ):
+        self.repo = repo
+        self.cache = None
+
+    def __call__(self) -> dict:
+        if self.cache is not None:
+            return self.cache
+
+        runs_factory = self.repo.run_metadata_db.runs()
+        query_results = defaultdict(SafeNone)
+        for run in runs_factory:
+            query_results[run.hash] = {
+                'experiment': run.experiment,
+                'tags': run.tags
+            }
```

Comment:
@mihran113 I think we have to cache an entire run object, since we might have checks like `run.name == '...'` or `run.description == '...'`